fem/qfunction.cpp

```diff
@@ -240,4 +242,45 @@ void QuadratureFunction::SaveVTU(const std::string &filename, VTKFormat format,
    SaveVTU(f, format, compression_level, field_name);
 }
 
+static double ReduceDouble(const Mesh *mesh, double value)
+{
+#ifdef MFEM_USE_MPI
+   if (auto *pmesh = dynamic_cast<const ParMesh*>(mesh))
+   {
+      MPI_Comm comm = pmesh->GetComm();
+      MPI_Allreduce(MPI_IN_PLACE, &value, 1, MPI_DOUBLE, MPI_SUM, comm);
+   }
+#endif
+   return value;
+}
+
+double QuadratureFunction::Integrate() const
+{
+   MFEM_VERIFY(vdim == 1, "Only scalar functions are supported.")
+   const double local_integral = (*this)*qspace->GetWeights();
```

Comment:
I know this isn't relevant to this PR, but I just want to say that Vector::operator*() really makes initially reviewing this code harder. It would have been nice if it had been called `InnerProduct` instead just for code readability. It took me looking up the Vector documentation to understand what the code was actually doing.

Reply:
Good point. There actually is a free function called `InnerProduct`, I switched to that instead of `operator*` in 4bfe05a9a5fd7788e022346eaf56c1a6354cc50c.